### PR TITLE
[fix] regex description for routes

### PR DESCRIPTION
**Pull request by Guaris**
### Summary
expresses that regex values need a ~ to be considered regex, and also that not including it means the route gets parsed as plain text. 

### Reason
reported. 

I've also created [docu-2630] for updating this entire doc.

### CODE REVIEW

File: src/gateway/how-kong-works/routing-traffic.md [hunk 1]
```diff
@@ -495,87 +495,38 @@ This allow you to define two routes with two paths: `/service` and
...
-and replaces with more direct approaches, but there are more complex expressions
-that could result in this scenario.
-
-To limit the worst-case scenario, {{site.base_gateway}} applies the OpenResty
-[`lua_regex_match_limit`](https://github.com/openresty/lua-nginx-module#lua_regex_match_limit)
-to ensure that any regex operation terminates in around two seconds in the
-worst case.  Apart from that, a smaller limit is applied to some "critical"
-regex operations, like those for path selection, in order to terminate them
-within two milliseconds, at most.
+For a deeper look into how the router processes regular expressions, read [performance considerations when using Expressions](/gateway/latest/key-concepts/routes/expressions/#performance-considerations-when-using-expressions).
```
Comment (cloudjumpercat) [inline src/gateway/how-kong-works/routing-traffic.md:509]:
```suggestion
For more information about how the router processes regular expressions, see [performance considerations when using Expressions](/gateway/latest/key-concepts/routes/expressions/#performance-considerations-when-using-expressions).
```
Could just be a personal preference, "deeper look" sounds kind of marketing-y to me

File: src/gateway/how-kong-works/routing-traffic.md [hunk 2]
```diff
@@ -495,87 +495,38 @@ This allow you to define two routes with two paths: `/service` and
...
 
 ##### Evaluation order
 
-As previously mentioned, {{site.base_gateway}} evaluates prefix paths by length, the longest
-prefix paths are evaluated first. However, {{site.base_gateway}} will evaluate Regex paths based
-on the `regex_priority` attribute of routes from highest priority to lowest.
-Regex paths are furthermore evaluated before prefix paths.
-
-Consider the following routes:
+The router evaluates routes using the `priority` field of the `Route` where a route is configured.
```
Comment (cloudjumpercat) [inline src/gateway/how-kong-works/routing-traffic.md:513]:
```suggestion
The router evaluates routes using the `regex_priority` field of the `Route` where a route is configured.
```
Not sure if that's accurate. I only see a `regex_priority` field in the example below, not a `priority` field.

File: src/gateway/how-kong-works/routing-traffic.md [hunk 3]
```diff
@@ -588,10 +539,8 @@ defined URIs, in this order:
 3. `/version/any/`
 4. `/version`
 
-Take care to avoid writing Regex rules that are overly broad and may consume
-traffic intended for a prefix rule. Adding a rule with the path `/version/.*` to
-the ruleset above would likely consume some traffic intended for the `/version`
-prefix path. If you see unexpected behavior, sending `Kong-Debug: 1` in your
+Routers that employ a large amount of regexes can consume traffic intended for other rules.
```
Comment (cloudjumpercat) [inline src/gateway/how-kong-works/routing-traffic.md:542]:
```suggestion
Routers with a large number of regexes can consume traffic intended for other rules.
```
Is this a good thing or bad thing? It almost feels like it's missing a "This results in...."

**Review by cloudjumpercat — CHANGES_REQUESTED** [review]
Just a few small suggestions, but the output looks good to me!

File: src/gateway/how-kong-works/routing-traffic.md (hunk 3 above)
Comment (Guaris) [reply src/gateway/how-kong-works/routing-traffic.md:542]: bad thing, should I make it more clear?

**Review by cloudjumpercat — APPROVED** [review]
LGTM! Also, nice job cutting out some of the words; makes it easier to understand.

**Pull request merged**